src/components/ui/popup/EnhancedPopper.tsx

```diff
@@ -0,0 +1,64 @@
+import { Placement } from 'popper.js';
+import * as React from 'react';
+import * as enhanceWithClickOutside from 'react-click-outside';
+import styled from 'react-emotion';
+import { Popper } from 'react-popper';
+
+export interface PopperProps {
+  backgroundColor?: string;
+  children: React.ReactNode;
+  onClickOutside?: () => void;
+  PopperComponent: typeof Popper;
+  placement?: Placement;
+  shadow?: boolean;
+  targetMinHeight?: number;
+  targetMinWidth?: number;
+}
+
+const TargetContainer = styled('div')<Partial<PopperProps>>(props => ({
+  backgroundColor: props.backgroundColor || '#fff',
+  boxShadow: props.shadow ? '0 0 8px 0 rgba(0, 0, 0, 0.15)' : undefined,
```

Comment:
Should reuse styling from Shadow component. So we don't have to maintain several definitions of the same style.